cluster/gce/util.sh

```diff
@@ -167,13 +188,26 @@ function upload-server-tars() {
 
   local -r staging_path="${staging_bucket}/devel"
 
+  local server_hash=$(sha1sum-file ${SERVER_BINARY_TAR})
+  local salt_hash=$(sha1sum-file ${SALT_TAR})
+
   echo "+++ Staging server tars to Google Storage: ${staging_path}"
   local server_binary_gs_url="${staging_path}/${SERVER_BINARY_TAR##*/}"
-  gsutil -q -h "Cache-Control:private, max-age=0" cp "${SERVER_BINARY_TAR}" "${server_binary_gs_url}"
-  gsutil acl ch -g all:R "${server_binary_gs_url}" >/dev/null 2>&1
+  if already-staged ${SERVER_BINARY_TAR} ${server_hash}; then
+    echo "+++ $(basename ${SERVER_BINARY_TAR}) already staged ('rm ${SERVER_BINARY_TAR}.sha1' to force)"
+  else
+    echo ${server_hash} > "${SERVER_BINARY_TAR}.sha1"
+    gsutil -m -q -h "Cache-Control:private, max-age=0" cp "${SERVER_BINARY_TAR}" "${SERVER_BINARY_TAR}.sha1" "${staging_path}"
+    gsutil -m acl ch -g all:R "${server_binary_gs_url}" "${server_binary_gs_url}.sha1" >/dev/null 2>&1
+  fi
   local salt_gs_url="${staging_path}/${SALT_TAR##*/}"
-  gsutil -q -h "Cache-Control:private, max-age=0" cp "${SALT_TAR}" "${salt_gs_url}"
-  gsutil acl ch -g all:R "${salt_gs_url}" >/dev/null 2>&1
+  if already-staged ${SALT_TAR} ${salt_hash}; then
```

Comment:
There's some redundancy in these two blocks, copying the server_binary and the salt... Should you maybe refactor that into its own function and call it twice here instead?